tir/tirClient.go

```diff
@@ -80,12 +83,20 @@ func NewTirHttpClient(tokenProvider TokenProvider, config config.M2M) (client Ti
 	_, err = httpcache.NewWithInmemoryCache(httpClient, true, time.Second*60)
 	if err != nil {
 		logging.Log().Errorf("Was not able to inject the cache to the client. Err: %v", err)
-		return
+		return nil, err
 	}
 	var httpGetClient HttpGetClient
 	if config.AuthEnabled {
-		logging.Log().Debug("Authorization for the tursted-issuers-registry is enabled.")
-		httpGetClient = AuthorizingHttpClient{httpClient: httpClient, tokenProvider: tokenProvider, clientId: config.ClientId}
+		logging.Log().Debug("Authorization for the trusted-issuers-registry is enabled.")
+		authorizingHttpClient := AuthorizingHttpClient{httpClient: httpClient, tokenProvider: tokenProvider, clientId: config.ClientId}
+
+		_, err := chrono.NewDefaultTaskScheduler().ScheduleAtFixedRate(authorizingHttpClient.FillMetadataCache, time.Duration(30)*time.Second)
```

Comment:
Should we link the refresh time to the cache expiration in some way?

Reply:
I took this piece of code from credentialsConfig.go, so, I think, it'll work the same as there